### PR TITLE
[repo] Remove .NET 7 target

### LINKED ISSUE #5639: Remove .NET 7 TFM
.NET 7 already reached EOL https://devblogs.microsoft.com/dotnet/dotnet-7-end-of-support/ on May 14, 2024, need to remove the TFM.

### PR DESCRIPTION
**Pull request by xiang17**
Fixes #5639.
Design discussion issue #5639.

## Changes

Please provide a brief description of the changes here.

## Merge requirement checklist

* [x] [CONTRIBUTING](https://github.com/open-telemetry/opentelemetry-dotnet/blob/main/CONTRIBUTING.md) guidelines followed (license requirements, nullable enabled, static analysis, etc.)
* [x] Unit tests added/updated
* [x] Appropriate `CHANGELOG.md` files updated for non-trivial changes
~~* [ ] Changes in public API reviewed (if applicable)~~

### CODE REVIEW

File: src/OpenTelemetry.Api/CHANGELOG.md [hunk 1]
```diff
@@ -14,6 +14,9 @@ Notes](../../RELEASENOTES.md).
 * Optimize performance of `TraceContextPropagator.Extract`.
   ([#5749](https://github.com/open-telemetry/opentelemetry-dotnet/pull/5749))
 
+* Remove .NET 7 TargetFramework which reached End of Support
```
Comment (xiang17) [inline src/OpenTelemetry.Api/CHANGELOG.md:17]: Shall I added an entry in each package's CHANGELOG.md file?

Comment (vishweshbankwar) [reply src/OpenTelemetry.Api/CHANGELOG.md:17]: we don't target `net7.0` for api and sdk.

File: .github/workflows/ci.yml [hunk 2]
```diff
@@ -129,7 +129,7 @@ jobs:
     strategy:
       fail-fast: false
       matrix:
-        version: [ net6.0, net7.0 ]
+        version: [ net6.0 ]
```
Comment (vishweshbankwar) [inline .github/workflows/ci.yml:132]: I think we should add `net8.0` here. Not sure why it is missing.

Comment (xiang17) [reply .github/workflows/ci.yml]: The Docker setup [failed](https://github.com/open-telemetry/opentelemetry-dotnet/actions/runs/10480814956/job/29029133563?pr=5795) for net8.0 after adding this. Not sure if it ever worked for .NET 8.

Comment (xiang17) [reply .github/workflows/ci.yml]: I tried to fix it with `venv`: https://stackoverflow.com/questions/75602063/pip-install-r-requirements-txt-is-failing-this-environment-is-externally-mana

But it's not quite trivial.

I think it's better to add in another PR. I'll revert adding net8.0 for this step and keep this PR simple.

Comment (vishweshbankwar) [reply .github/workflows/ci.yml:132]: could you please open an issue to track?

**Review by vishweshbankwar — APPROVED** [review]

File: .github/workflows/ci.yml (hunk 2 above)
Comment (xiang17) [reply .github/workflows/ci.yml:132]: Opened an issue: #5799.

**Pull request merged**